src/sb3-serialize.ts

```diff
@@ -223,15 +225,42 @@ function deserializeBlockObject(serialized: SB3SerializedBlockObject, id: string
     return [block, ...newBlocks];
 }
 
-export function deserializeVariables(serialized: SB3Variables): Iterator<VariableProperties> {
+export function deserializeVariables(serialized: SB3ScalarVariables): Iterator<VariableProperties> {
     return map(
         Object.entries(serialized),
         ([id, [name, value, isCloud]]): VariableProperties => ({
             id,
             name,
             value,
             isCloud: Boolean(isCloud), //when  isCloud is undefined
-            type: '', // this means scalar type in https://github.com/LLK/scratch-vm/blob/33ef283787d4ea9a90c3d0d069a6b97dee24f51b/src/engine/variable.js#L49
+            type: ScratchVariableTypes.SCALAR_TYPE, // this means scalar type in https://github.com/LLK/scratch-vm/blob/33ef283787d4ea9a90c3d0d069a6b97dee24f51b/src/engine/variable.js#L49
```

Comment:
thanks for linking that - its super helpful! Alot doc tools (link jsdocs) support `@link` for external documention https://jsdoc.app/tags-inline-link.html 

I'm hoping the TypeScript docs support that as well..